src/unix/syscall.c

```diff
@@ -527,6 +529,7 @@ closure_function(9, 2, void, sendfile_bh,
         if (bound(cur_buf)->misc == bound(cur_buf)->length) {
             sg_buf_release(bound(cur_buf));
             if (bound(written) == bound(readlen)) {
+                bound(cur_buf) = 0;
                 rv = bound(written);
                 goto out_complete;
```

Comment:
This doesn't seem necessary, since at this point we are about to exit from the syscall, so `cur_buf` will never be used again.